nima/http2/webserver/src/main/java/io/helidon/nima/http2/webserver/Http2Config.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.nima.http2.webserver;
+
+import io.helidon.config.Config;
+import io.helidon.nima.http2.Http2Setting;
+import io.helidon.nima.http2.Http2Settings;
+
+/**
+ * HTTP/2 server configuration.
+ */
+class Http2Config {
+
+    private static final String CONFIG_MAX_FRAME_SIZE = "max-frame-size";
+    private static final String CONFIG_MAX_HEADER_LIST_SIZE = "max-header-size";
+
+    private final Long maxFrameSize;
+    private final Long maxHeaderListSize;
+
+    private Http2Config(Long maxFrameSize, Long maxHeaderListSize) {
```

Comment:
use primitive types, this must not be null

Reply:
fixed